kolibri/plugins/coach_tools/assets/src/actions.js

```diff
@@ -109,34 +93,33 @@ function showReport(store, params, oldParams) {
 
 
   /* check if params are semi-valid. */
-  if (!(_vals(Constants.ContentScopes).includes(contentScope)
-    && _vals(Constants.UserScopes).includes(userScope)
-    && _vals(Constants.AllOrRecent).includes(allOrRecent)
-    && _vals(Constants.ViewBy).includes(viewByContentOrLearners)
-    && _vals(Constants.SortCols).includes(sortColumn)
-    && _vals(Constants.SortOrders).includes(sortOrder))) {
+  if (!(Constants.enumerate(Constants.ContentScopes).includes(contentScope)
+    && Constants.enumerate(Constants.UserScopes).includes(userScope)
+    && Constants.enumerate(Constants.AllOrRecent).includes(allOrRecent)
+    && Constants.enumerate(Constants.ViewBy).includes(viewByContentOrLearners)
+    && Constants.enumerate(Constants.TableColumns).includes(sortColumn)
+    && Constants.enumerate(Constants.SortOrders).includes(sortOrder))) {
     /* if invalid params, just throw an error. */
     coreActions.handleError(store, 'Invalid report parameters.');
     return;
   }
 
+  const diffKeys = _diffKeys(params, oldParams);
+
   store.dispatch('SET_PAGE_NAME', Constants.PageNames.REPORTS);
 
-  if (_useReportPageLoadingFlag(params, oldParams)) {
-    store.dispatch('CORE_SET_PAGE_LOADING', true);
+  // these don't require updates from the server
+  const localUpdateParams = ['sort_column', 'sort_order'];
+  if (diffKeys.every(key => localUpdateParams.includes(key))) {
+    store.dispatch('SET_SORT_COLUMN', sortColumn);
+    store.dispatch('SET_SORT_ORDER', sortOrder);
+    return;
   }
 
-  /* save all params to store. */
-  store.dispatch('SET_CHANNEL_ID', channelId);
-  store.dispatch('SET_CONTENT_SCOPE', contentScope);
-  store.dispatch('SET_CONTENT_SCOPE_ID', contentScopeId);
-  store.dispatch('SET_USER_SCOPE', userScope);
-  store.dispatch('SET_USER_SCOPE_ID', userScopeId);
-  store.dispatch('SET_ALL_OR_RECENT', allOrRecent);
-  store.dispatch('SET_VIEW_BY_CONTENT_OR_LEARNERS', viewByContentOrLearners);
-  store.dispatch('SET_SORT_COLUMN', sortColumn);
-  store.dispatch('SET_SORT_ORDER', sortOrder);
-
+  // set loading true for all except 'view-by', which just affects table (not summary)
+  if (diffKeys.length === 1 && diffKeys[0] !== 'view_by_content_or_learners') {
```

Comment:
The `=== 1` is actually necessary to ensure that _only_ that key changed. I'll update the comment to clarify.